.github/workflows/ccpp.yml

```diff
@@ -38,40 +38,25 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: '3.12'
+
       - name: Install linting tools
         run: pip install clang-format cmake-format[YAML] mypy ruff
-      - name: ruff .py files in C++ code
-        run: |
-          cd cpp/
-          ruff check .
-          ruff format --check .
-      - name: ruff Python interface checks
-        run: |
-          cd python/
-          ruff check .
-          ruff format --check .
-      - name: mypy checks
+
+      - name: Run ruff
+        run: ruff check python/ cpp/
+
+      - name: Run mypy
+        working-directory: python/
         run: |
-          cd python/
           mypy dolfinx
           mypy demo
           mypy test
-      - name: clang-format C++ checks (non-blocking)
-        continue-on-error: true
-        run: |
-          cd cpp
-          clang-format --version
-          find . -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
-      - name: clang-format Python binding checks (non-blocking)
-        continue-on-error: true
-        run: |
-          cd python/dolfinx/wrappers
-          clang-format --version
-          find . -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
-      - name: cmake-format (non-blocking)
-        continue-on-error: true
-        run: |
-          find . -type f \( -name "*.cmake" -o -name "*.cmake.in" -o -name "CMakeLists.txt" \) | xargs cmake-format --check
+
+      - name: Run clang-format
+        run: find cpp/ python/dolfinx/wrappers/ -type f \( -name "*.cpp" -o -name "*.h" \) | xargs clang-format --dry-run --Werror
```

Comment:
Split into two steps is intended so it is clear which part of the library is the issue.